applications/views.py

```diff
@@ -1,15 +1,31 @@
 from django.contrib.auth.decorators import login_required
+from django.http import Http404
 from django.shortcuts import get_object_or_404, redirect
 from django.template.response import TemplateResponse
 
+from jobserver.authorization import has_permission
+
 from .form_specs import form_specs
 from .models import Application
 from .wizard import Wizard
 
 
+def validate_application_access(user, application):
+    if has_permission(user, "application_manage"):
+        return
+
+    if application.created_by == user:
+        return
+
+    raise Http404
+
+
 @login_required
 def page(request, pk, page_num):
     application = get_object_or_404(Application, pk=pk)
+
+    validate_application_access(request.user, application)
```

Comment:
Should we add this to the confirmation view too?

Reply:
Yes, oversight on my part 👍